ocaml/tests/test_clustering.ml

```diff
@@ -12,249 +12,259 @@
  * GNU Lesser General Public License for more details.
  *)
 
-(** Tests for the Xapi_clustering module *)
+(** Test for the Xapi_clustering module *)
 
 module T = Test_common
 
-(** Tests Xapi_clustering.get_required_cluster_stacks *)
-let test_get_required_cluster_stacks =
-  let assert_equal ~required_cluster_stacks ~to_set =
-    let module S = Ounit_comparators.StringSet in
-    S.assert_equal (S.of_list to_set) (S.of_list required_cluster_stacks)
-  in
-
-  let test_zero_sms_in_database () =
-    let __context = T.make_test_database () in
-    let required_cluster_stacks = Xapi_clustering.get_required_cluster_stacks ~__context ~sr_sm_type:"" in
-    assert_equal ~required_cluster_stacks ~to_set:[]
-  in
+(** Test Xapi_clustering.get_required_cluster_stacks *)
+
+let assert_equal ~msg ~required_cluster_stacks ~to_set =
+  let module S = Set.Make(String) in
+  let setify sl = sl |> S.of_list |> S.elements in
+  Alcotest.(check (slist string String.compare)) msg
+    (setify required_cluster_stacks) (setify to_set)
```

Comment:
@edwintorok Is it okay that `get_required_cluster_stacks` might return repeated elements, or should we put this into the code itself instead of the test?

Reply:
I think that `get_required_cluster_stacks` should return either an empty list, or a list of one element. If it returns more than one element that is not going to work: different cluster stacks are mutually incompatible.
So perhaps we should fix the code to return an option, and raise an exception if the list ever contains more than 2 elements.

Reply:
I think the way this function was originally intended is that this is a disjunction not a conjunction: we require one of these cluster stacks for the SM to work, not all. So maybe this function assumes that there will be SMs in the future that work with multiple different cluster stacks.